### PR TITLE
JDK-8314611: Provide more explicative error message parsing Currencies

### PR DESCRIPTION
**Pull request by justin-curtis-lu**
Please review this PR which updates some exceptions in j.util.Currency to have an explicit error message (as opposed to nothing).

The exceptions are thrown when the ISO 4217/3166 currency/country codes are in an invalid form, or do not exist.

---------
### Progress
- [x] Change must be properly reviewed (1 review required, with at least 1 [Reviewer](https://openjdk.org/bylaws#reviewer))
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue

### Issue
 * [JDK-8314611](https://bugs.openjdk.org/browse/JDK-8314611): Provide more explicative error message parsing Currencies (**Enhancement** - P4)

### Reviewers
 * [Naoto Sato](https://openjdk.org/census#naoto) (@naotoj - **Reviewer**)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/jdk.git pull/15458/head:pull/15458` \
`$ git checkout pull/15458`

Update a local copy of the PR: \
`$ git checkout pull/15458` \
`$ git pull https://git.openjdk.org/jdk.git pull/15458/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 15458`

View PR using the GUI difftool: \
`$ git pr show -t 15458`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/jdk/pull/15458.diff">https://git.openjdk.org/jdk/pull/15458.diff</a>

</details>

### Webrev
[Link to Webrev Comment](https://git.openjdk.org/jdk/pull/15458#issuecomment-1696431200)

### CODE REVIEW

File: src/java.base/share/classes/java/util/Currency.java [hunk 1]
```diff
@@ -342,7 +343,8 @@ private static Currency getInstance(String currencyCode, int defaultFractionDigi
             if (!found) {
                 OtherCurrencyEntry ocEntry = OtherCurrencyEntry.findEntry(currencyCode);
                 if (ocEntry == null) {
-                    throw new IllegalArgumentException();
+                    throw new IllegalArgumentException("The input currency code" +
+                            " is not a supported ISO 4217 code");
```
Comment (naotoj) [inline src/java.base/share/classes/java/util/Currency.java:347]: I think "valid" is more appropriate than "supported." Applies to other locations as well.

**Review by naotoj — COMMENTED** [review]
Looks good. Minor suggestion to the exception message.

**Comment by justin-curtis-lu** [comment]
> I think the same wording changes are needed in the test, as it checks the error messages word by word.

You're totally right, just fixed.

**Pull request closed without merging**